### PR TITLE
Work in progress - Fixes for Mongoose5.0.2

### PR DESCRIPTION
**Pull request by alexpchin**
Removing this prevents the error as mentioned in: #7.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -36,20 +36,21 @@ const counterSchema = new mongoose.Schema({
...
-  {
-    field: 1,
-    groupingField: 1,
-    model: 1,
-  },
-  {
-    unique: true,
-  }
-);
+// Deprecated after upgrading to Mongoose5.
```
Comment (sp3c1) [inline src/index.js:39]: It  made the error not get triggered, but its aint a solution ;]. Its not deprecated as such.  Have looked back at the 5.0.2 doc. 
```
animalSchema.index({ name: 1, type: -1 }); 
```
should be still valid ^^

ps. guess you are working on it but maybe lets mark the PR as WIP? ;]

**Comment by alexpchin** [comment]
Oh yeah, I forgot to do that. Was just working through the issue to discover the part that caused the error ;)

**Comment by nodkz** [comment]
Identitycounters collection must have unique index. Commenting it out, does not solve root of problem. Try to look what happens tommorow at work.

**Comment by alexpchin** [comment]
I was meant to make this as "work in progress".

**Comment by nodkz** [comment]
@alexpchin can you add your test case to this PR?

You'll see how tomorrow i modify it to work with different mongoose version with mongodb-memory-server with one tests run.

**Comment by nodkz** [comment]
Sorry for delay.

I close this PR due new tests and code changes in https://github.com/nodkz/mongoose-plugin-autoinc/commit/d256a8449b6dee5a443a3ace40c2a83d82ed0ec1

Feel free to open new PR if needed. 
Continue discussion in #7

**Pull request closed without merging**